api/src/test/java/org/apache/iceberg/metrics/TestDefaultTimer.java

```diff
@@ -104,7 +106,7 @@ public void measureRunnable() {
     Runnable runnable =
         () -> {
           try {
-            Thread.sleep(100);
+            Awaitility.await().atLeast(Duration.ofMillis(100)).until(() -> true);
```

Comment:
won't it become flaky, if callable condition is evaluated sooner. 
`if (evaluationDuration.compareTo(minWaitTime) < 0) {
                String message = String.format("Condition was evaluated in %s which is earlier than expected minimum timeout %s",
                        formatAsString(evaluationDuration), formatAsString(minWaitTime));
                conditionEvaluationHandler.handleTimeout(message, true);
                throw new ConditionTimeoutException(message);
            }`
            
            Why not use something like `https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/TestHelpers.java#L57`
            ?